x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/config/AnalysisConfig.java

```diff
@@ -146,8 +135,13 @@ public AnalysisConfig(StreamInput in) throws IOException {
         summaryCountFieldName = in.readOptionalString();
         detectors = Collections.unmodifiableList(in.readList(Detector::new));
         influencers = Collections.unmodifiableList(in.readList(StreamInput::readString));
-        overlappingBuckets = in.readOptionalBoolean();
-        resultFinalizationWindow = in.readOptionalLong();
+
+        // BWC for result_finalization_window and overlapping_buckets
+        // TODO Remove in 7.0.0
+        if (in.getVersion().before(Version.V_6_6_0)) {
```

Comment:
This is going to make the CI for this PR fail. It's actually quite complex how to deal with this without breaking the build. I'll reach out to you to explain.

Reply:
Yeah, I noticed some previous pains you experience when handling the the multiple bucket span options...Would like to learn the best path forward :D.